### PR TITLE
Support Unicode 8 in System.Text.Encodings.Web

### PR DESCRIPTION
**Pull request by tarekgh**
The change here is to upgrade the Unicode data used in System.Text.Encodings.Web
from Unicode version 7.0 to 8.0. Also the change fixes the issue in the data
which we used to ignore the codepoints inside the spans like:

   3400;<CJK Ideograph Extension A, First>;Lo;0;L;;;;;N;;;;;
   4DB5;<CJK Ideograph Extension A, Last>;Lo;0;L;;;;;N;;;;;

But now with the change,  all codepoints inside the spans are included if it has
one of the supported category

Note: We need to port the tool that generate the Unicode data to somewhere.
Currently we don’t include any tools in corefx repo so we need to find a home
for such tool (maybe corefx-tools repo?).

### CODE REVIEW

**Comment by tarekgh** [comment]
CC: @KrzysztofCwalina @Eilon @GrabYourPitchforks 
and link the PR to the issue #4443

**Comment by KrzysztofCwalina** [comment]
Looks good to me, but I swear when I looked at @GrabYourPitchforks's changes there were more of them :-)

File: src/System.Text.Encodings.Web/tests/UnicodeHelpersTests.cs [hunk 1]
```diff
@@ -178,10 +179,24 @@ private static bool[] ReadListOfDefinedCharacters()
...
                         retVal[codePoint] = true; // chars in this category are allowable
                         seenCategories.Add(category);
+                        
+                        if (splitLine[1].EndsWith("First>"))
+                        {
+                            startSpanCodepoint = codePoint;
+                        } 
+                        else if (splitLine[1].EndsWith("Last>"))
+                        {
+                            for (uint spacCounter = startSpanCodepoint; spacCounter < codePoint; spacCounter++)
```
Comment (Eilon) [inline src/System.Text.Encodings.Web/tests/UnicodeHelpersTests.cs]: `spaceCounter` :wink:

Comment (tarekgh) [inline src/System.Text.Encodings.Web/tests/UnicodeHelpersTests.cs]: thanks for the catch :-)

Comment (Eilon) [inline src/System.Text.Encodings.Web/tests/UnicodeHelpersTests.cs]: Ah, duh :smile:

**Comment by tarekgh** [comment]
The failed test is unrelated and I have logged a bug #4555 for this failure. 

@GrabYourPitchforks, I am going to merge the changes but if you get any comment please let me know and we can fix it.

**Pull request merged**